group/group.go

```diff
@@ -0,0 +1,74 @@
+package group
+
+import (
+	"crypto/elliptic"
+	"encoding"
+	"errors"
+	"io"
+
+	"github.com/cloudflare/circl/ecc/p384"
+)
+
+var (
+	// P256 is the group generated by P-256 elliptic curve.
+	P256 Group = wG{elliptic.P256()}
+	// P384 is the group generated by P-384 elliptic curve.
+	P384 Group = wG{p384.P384()}
+	// P521 is the group generated by P-521 elliptic curve.
+	P521 Group = wG{elliptic.P521()}
+)
+
+// Group si
+type Group interface {
+	NewElt() Element
```

Comment:
While I like the simplicity of this, it might be better to actually say 'NewElement'. The same for 'NewScalar'.